Sources/Task/ExistentialTask.swift

```diff
@@ -91,14 +91,16 @@ extension Task {
         self.init(future: base, progress: progress)
     }
 
+    private typealias _Self = Task<SuccessValue>
+
     /// Create a task whose `upon(_:_:)` method uses the result of `base`.
     ///
     /// If `base` is not a `Task`, `cancellation` will be called asynchronously,
     /// but not on any specific queue. If you must do work on a specific queue,
     /// schedule work on it.
     public convenience init<Task: FutureType where Task.Value: ResultType, Task.Value.Value == SuccessValue>(_ base: Task, cancellation: ((Void) -> Void)? = nil) {
-        let progress = NSProgress.wrapped(base, cancellation: cancellation)
-        self.init(future: Future(task: base), progress: progress)
+        let underlying = (base as? _Self)?.progress ?? NSProgress.wrapped(base, cancellation: cancellation)
```

Comment:
It wasn't clear to me from the docs that the cancellation will _only_ be called if `base` is not a Task delivering the same value as `Self`. That would surprise me for sure as a dev, and I'd probably mess that up at some point. :\ Is it possible to overload such that we don't take the `cancellation` param unless we're going to use it?